torchgeo/datasets/cdl.py

```diff
@@ -43,23 +43,23 @@ class CDL(RasterDataset):
     is_image = False
```

Comment:
I think the glob/regex will still pick up versions other than the ones requested if they are found on the system. You _could_ fix this by editing the `filename_regex` during `__init__`. We do something similar to this in our Landsat datasets where we patch `filename_glob` based on the bands requested.